lib/WUI/nhttp/send_json.cpp

```diff
@@ -0,0 +1,124 @@
+#include "send_json.h"
+#include "handler.h"
+#include "headers.h"
+
+#include <timing.h>
+#include <segmented_json_macros.h>
+#include <http/chunked.h>
+#include <transfers/monitor.hpp>
+
+using transfers::Monitor;
+using namespace json;
+using namespace http;
+
+namespace nhttp::handler {
+
+JsonResult TransferRenderer::renderState(size_t resume_point, json::JsonOutput &output, TransferState &state) const {
+    auto transfer_status = Monitor::instance.status(resume_point != 0);
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+        JSON_OBJ_START
+            if (transfer_status.has_value()) {
+                if (resume_point != 0 && transfer_status->id != state.transfer_id)
+                    // if transfer changes mid report, bail out
+                    transfer_status.reset();
+                else
+                    state.transfer_id = transfer_status->id;
+
+                JSON_FIELD_STR("target", "usb") JSON_COMMA;
+                JSON_FIELD_STR_FORMAT_G(transfer_status.has_value(), "type", "%s", to_str(transfer_status->type)) JSON_COMMA;
+                JSON_FIELD_STR_FORMAT_G(transfer_status.has_value(), "size", "%d", transfer_status->expected) JSON_COMMA;
+                //FIXME: Right now should be seconds from epoch start, would be much nicer
+                // for us to make it seconds from print start and make the client compute the rest,
+                // also it would solve some potential problems with NTP and time zones etc.
+                JSON_FIELD_STR_FORMAT_G(transfer_status.has_value(), "start_time", "%d", time(nullptr) - (ticks_s() - transfer_status->start)) JSON_COMMA;
+                // FIXME: Does this make sense in case of connect? Probably should be added
+                // once we actually have a working connect transfer implementation.
```

Comment:
No, connect doesn't have any notion of „Transfer & print imediatelly“. The server keeps track of these things and would send a separate command to start printing once the transfer is finished.

Feel free to add such flags into the monitor as needed.